examples/chatgpt/gpt_actions_library/gpt_action_sharepoint_doc.ipynb

```diff
@@ -0,0 +1,640 @@
+    "               for (const hit of container.hits) {\n",
+    "                   if (hit.resource[\"@odata.type\"] === \"#microsoft.graph.driveItem\") {\n",
+    "                       const { name, id } = hit.resource;\n",
+    "                       // The below is where the file lives\n",
+    "                       const driveId = hit.resource.parentReference.driveId;\n",
+    "                       // we use the helper function we defined above to get the contents, convert to base64, and restructure it\n",
+    "                       const contents = await getDriveItemContent(client, driveId, id, name);\n",
+    "                       results.push(contents)\n",
+    "               }\n",
+    "           }));\n",
+    "           return results;\n",
+    "       };\n",
+    "       let results;\n",
+    "       if (list.value[0].hitsContainers[0].total == 0) {\n",
+    "           // Return no results found to the API if the Microsoft Graph API returns no results\n",
+    "           results = 'No results found';\n",
+    "       } else {\n",
+    "           // If the Microsoft Graph API does return results, then run processList to iterate through.\n",
+    "           results = await processList();\n",
+    "           // this is where we structure the response so ChatGPT knows they are files\n",
+    "           results = {'openaiFileResponse': results}\n",
+    "       }\n",
+    "       context.res = {\n",
+    "           status: 200,\n",
+    "           body: results\n",
+    "       };\n",
+    "   } catch (error) {\n",
+    "       context.res = {\n",
+    "           status: 500,\n",
+    "           body: `Error performing search or processing results: ${error.message}`,\n",
+    "       };\n",
+    "   }\n",
+    "};\n",
+    "```\n",
+    "### Customizations\n",
+    "\n",
+    "Below are some potential areas to customize. \n",
+    "\n",
+    "- You can customize the GPT prompt to search again a certain amount of times if nothing is found.\n",
+    "\n",
+    "- You can customize the code to only search through specific SharePoint sites or O365 Drives by customizing the search query. This will help focus the search and improve the retrieval. The function as setup now looks through all files the logged-in user can access.\n",
+    "\n",
+    "- You can update the code to only return certain types of files. For example, only return structured data / CSVs. \n",
+    "\n",
+    "- You can customize the amount of files it searches through within the call to Microsoft Graph. Note that you should only put a maximum of 10 files based on the documentation [here](https://platform.openai.com/docs/actions/getting-started). \n",
+    "\n",
+    "### Considerations\n",
+    "\n",
+    "Note that all the same limitations of Actions apply here, with regards to returning 100K characters or less and the [45 second timeout](https://platform.openai.com/docs/actions/production/timeouts).\n",
```

Comment:
this might be slightly confusing for the doc return part...since we are returning the doc and not plaintext charagers